wgengine/userspace.go

```diff
@@ -145,26 +145,31 @@ type EngineConfig struct {
 	// which disables such features as DNS configuration and unrestricted ICMP Echo responses.
 	Fake bool
 
-	// FakeImpl, if non-nil, specifies which type of fake implementation to
-	// use. Two values are typical: nil, for a basic ping-only fake
-	// implementation, and netstack.Impl, which brings in gvisor's netstack
-	// to the binary. The desire to keep that out of some binaries is why
-	// this func exists, so wgengine need not depend on gvisor.
-	FakeImpl FakeImplFunc
+	// FakeImplFactory, if non-nil, creates a FakeImpl to use as a fake
+	// implementation. See docs for FakeImpl below for more details.
+	FakeImplFactory FakeImplFactory
 }
 
-// FakeImplFunc is the type used by EngineConfig.FakeImpl. See docs there.
-type FakeImplFunc func(logger.Logf, *tstun.TUN, Engine, *magicsock.Conn) error
+// FakeImpl specifies which type of fake implementation to use. Two values
```

Comment:
did you see my earlier comment about these docs? this is the place to say what FakeImpl is/does, not what nil means.

Also it's not a func ("why this func exists").

How about just:

// FakeImpl is a fake or alternate version of Engine that can be started. See EngineConfig.FakeImplFactory.

Then restore some of the docs deleted on EngineConfig.FakeImpl to mention the implementations and the meaning of nil.

Reply:
I think I might have misunderstood that last comment. Should be better now.